pkgs/os-specific/linux/kernel/hardened/update.py

```diff
@@ -39,6 +39,16 @@
 })
 
 
+def read_min_kernel_branch():
+    with open(NIXPKGS_KERNEL_PATH / "kernels-org.json") as f:
+        components = list(map(
+            int,
+            sorted(json.loads(f.read()).keys())[0].split('.')
```

Comment:
Can do `[int(x) for x in ...]` and `json.load(f)` here for readability. Also maybe use `packaging.Version` to parse the version specifiers? Not sure if it's worth the dependency, this line just scares me.